packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -321,6 +322,43 @@ describe("BnsConnection", () => {
     connection.disconnect();
   });
 
+  it("can register a username", async () => {
+    pendingWithoutBnsd();
+    const connection = await BnsConnection.establish(bnsdTendermintUrl);
+    const chainId = await connection.chainId();
+
+    const profile = new UserProfile();
+    const wallet = profile.addWallet(Ed25519HdWallet.fromEntropy(await Random.getBytes(32)));
+    const identity = await profile.createIdentity(wallet.id, HdPaths.simpleAddress(0));
+
+    // Create and send registration
+    const address = keyToAddress(identity.pubkey);
+    const username = `testuser`; // _${Math.random()}`;
+    const registration: RegisterUsernameTx = {
+      kind: TransactionKind.RegisterUsername,
+      chainId: chainId,
+      signer: identity.pubkey,
+      addresses: new Map([[chainId, address]]),
+      username: username,
+    };
+    const nonce = await getNonce(connection, address);
+    const signed = await profile.signTransaction(wallet.id, identity, registration, bnsCodec, nonce);
+    const txBytes = bnsCodec.bytesToPost(signed);
+    await connection.postTx(txBytes);
+
+    // Find registration transaction
+    const searchResult = await connection.searchTx({ tags: [bnsFromOrToTag(address)] });
```

Comment:
Do not expect this to be found, this is only if the balance changes at the given address.

`bnsNonceTag` should find it, as it detects anything that was signed by that address